frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1720,9 +1731,16 @@ function ReaderFooter:setTocMarkers(reset)
         end
         self.progress_bar.ticks = {}
         if self.ui.toc then
-            self.progress_bar.ticks = self.ui.toc:getTocTicksFlattened()
+            -- filter the ticks to show only those in the current flow
+            for i,j in pairs(self.ui.toc:getTocTicksFlattened()) do
+                if self.view.document:getPageFlow(j) == self.flow then
+                    self.progress_bar.ticks[i] = self.view.document:getPageNumberInFlow(j)
+                else
+                    self.progress_bar.ticks[i] = nil
+                end
+            end
```

Comment:
In the same vein, only relevant if `hasFlows` (well, has and show, I guess, actually) ;).

Reply:
You *should* be able to iterate through `getTocTicksFlattened()` via `ipairs`. If it turns out the array is actually sparse, then we have a host of other buggy code that assumes it wouldn't ;).